File: Java/Practice/LeetCode/Easy/Maximum Sum With Exactly K Elements.java
// https://leetcode.com/problems/maximum-sum-with-exactly-k-elements/description/

class Solution {
    public int maximizeSum(int[] nums, int k) {
        Arrays.sort(nums);

        int kSum = 0;

        for (int i = 0; i < k; i++) {
            kSum += i;
        }
        
        return nums[nums.length - 1] * k + kSum;   
    }
}

File: TypeScript/Practice/LeetCode/Easy/Maximum Sum With Exactly K Elements.ts
// https://leetcode.com/problems/maximum-sum-with-exactly-k-elements/description/

function maximizeSum(nums: number[], k: number): number {
    nums.sort((a, b) => a - b);

    let kSum: number = 0;

    for (let i = 0; i < k; i++) {
        kSum += i;
    }
    
    return nums[nums.length - 1] * k + kSum;
};